Validator/Constraints/Recaptcha3.php

```diff
@@ -7,7 +7,9 @@
 /**
  * @Annotation
  */
-class Recaptcha3 extends Constraint
+final class Recaptcha3 extends Constraint
 {
+    const INVALID_FORMAT_ERROR = '7147ffdb-0af4-4f7a-bd5e-e9dcfa6d7a2d';
```

Comment:
Makes no sense to have it here! It's for testing purpose only.

Reply:
@Nek- it's used in the validator [test](https://github.com/karser/KarserRecaptcha3Bundle/blob/v0.1.2/Tests/Validator/Constraints/Recaptcha3ValidatorTest.php#L75)

Reply:
@karser that's what I said. You put it in a distributed class while it's only for testing purpose. Which could lead to wrong usage of a testing stuff. (well, it's about writing tests with tests and lib with lib)

Reply:
Hmm, I saw this pattern in the [symfony code](https://github.com/symfony/symfony/search?p=1&q=INVALID_FORMAT_ERROR&unscoped_q=INVALID_FORMAT_ERROR). Although I thought it is used in something else.
Well, isn't it what called testable code? When you change the design of the classes so it's easier to test. Anyways, I don't think it's a huge issue, isn't it?

Reply:
That's a good point - we could match the violation using only the `message`.
```
        $this->validator->validate($testToken, new Recaptcha3(['message' => 'myMessage']));

        $this->buildViolation('myMessage')
            ->setParameter('{{ value }}', '"'.$testToken.'"')
            ->setCode(Recaptcha3::INVALID_FORMAT_ERROR)
            ->assertRaised();
```

Reply:
In Symfony code, they are actually error code (represented by uuid because they are by definition unique). Not test stuff.

Reply:
Not fully following this: if I use the Symfony approach, what is the difference?

Reply:
You're **not** doing it the same way of Symfony. That's my point. Take a look at how they use this constant again.